tests/loopbacked_tests.rs

```diff
@@ -33,19 +33,53 @@ use util::simple_tests::test_setup;
 use util::simple_tests::test_teardown;
 use util::simple_tests::test_thinpool_device;
 
+pub struct LoopTestDev {
+    ld: LoopDevice,
+}
+
+impl LoopTestDev {
+    pub fn new(lc: &LoopControl, path: &std::path::PathBuf) -> LoopTestDev {
+        // Wipe 1 MiB at the beginning, as data sits around on the files.
+        OpenOptions::new()
+            .read(true)
+            .write(true)
+            .open(path)
+            .unwrap();
+
+        let ld = lc.next_free().unwrap();
+        ld.attach(path.clone(), 0).unwrap();
+        wipe_sectors(&ld.get_path().unwrap(),
+                     Sectors(0),
+                     Bytes(IEC::Mi as u64).sectors())
+                .unwrap();
+
+        LoopTestDev { ld: ld }
+    }
+
+    fn get_path(&self) -> PathBuf {
+        self.ld.get_path().unwrap()
+    }
+
+    pub fn detach(&self) -> Result<()> {
+        self.ld.detach()
+    }
+}
+
+impl Drop for LoopTestDev {
+    fn drop(&mut self) {
+        let _ = self.ld.detach();
+    }
+}
 
 /// Setup count loop backed devices in dir.
 /// Make sure each loop device is backed by a 1 GiB file.
 /// Wipe the first 1 MiB of the file.
-fn get_devices(count: u8, dir: &TempDir) -> Vec<LoopDevice> {
+fn get_devices(count: u8, dir: &TempDir) -> Vec<LoopTestDev> {
     let lc = LoopControl::open().unwrap();
     let mut loop_devices = Vec::new();
 
-    let length = Bytes(IEC::Gi as u64);
-    let wipe_length = Bytes(IEC::Mi as u64).sectors();
     for index in 0..count {
-        let subdir = TempDir::new_in(dir, &index.to_string()).unwrap();
-        let path = subdir.path().join("store");
+        let path = dir.path().join(format!("store{}", &index));
```

Comment:
I think I had my reasons, but it's quite possible that ```<tempdir>/0/store``` is no better than ```<tempdir>/store0```? Feel free to ignore this one.